context.go

```diff
@@ -141,6 +141,16 @@ func (c *Context) NumFlags() int {
 	return c.flagSet.NFlag()
 }
 
+// Set sets a context flag to a value.
+func (c *Context) Set(name, value string) error {
+	return c.flagSet.Set(name, value)
+}
+
+// GlobalSet sets a context flag to a value on the global flagset
+func (c *Context) GlobalSet(name, value string) error {
+	return globalContext(c).flagSet.Set(name, value)
```

Comment:
I think this may only want to walk up the tree as far as needed to find where the flag is defined (see `lookupGlobalFlagSet` e.g.).